x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/accesscontrol/wrapper/IndicesAccessControlWrapper.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authz.accesscontrol.wrapper;
+
+import org.elasticsearch.xpack.core.security.authz.accesscontrol.IndicesAccessControl;
+
+/**
+ * A wrapper which allows adding additional functionality and intercepting calls to {@link IndicesAccessControl} methods.
+ */
+@FunctionalInterface
+public interface IndicesAccessControlWrapper {
```

Comment:
An interface for a single class with a single method seems like overkill.

Reply:
There are two implementations: [anonymous](https://github.com/elastic/elasticsearch/blob/61e3075b77dbbe28b9f511c253622b759ea04a77/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L179) and `DlsFlsFeatureTrackingIndicesAccessControlWrapper`. I don't feel strongly with this approach. I'm okay to remove it.

Reply:
Removed https://github.com/elastic/elasticsearch/pull/88708/commits/d57f820122a39bb39cacdfda1ac88e6979edf28b